plugins/modules/lxc_container.py

```diff
@@ -518,48 +517,39 @@
 popd
 
 # User defined command
-%(container_command)s
+{}
 """
 
 
-def create_script(command):
+def create_script(command: str, module: AnsibleModule) -> None:
     """Write out a script onto a target.
 
     This method should be backward compatible with Python when executing
     from within the container.
 
     :param command: command to run, this can be a script and can use spacing
                     with newlines as separation.
-    :type command: ``str``
+    :param module: AnsibleModule to run commands with.
     """
 
-    (fd, script_file) = tempfile.mkstemp(prefix="lxc-attach-script")
-    f = os.fdopen(fd, "wb")
+    script_file = ""
     try:
-        f.write(to_bytes(ATTACH_TEMPLATE % {"container_command": command}, errors="surrogate_or_strict"))
+        f = tempfile.NamedTemporaryFile(prefix="lxc-attach-script", delete=False, mode="wb")
```

Comment:
Why not use a `with` statement for `f` (and remove `delete=False`)? After all you *do* want to delete the script later on.